src/index.ts

```diff
@@ -199,6 +210,21 @@ export class TopicProducer {
             throw new Error(`failed to send key-value record due to: ${error}`)
         }
     }
+
+    /**
+     * Sends a list of key-value elements to this producer's topic
+     * @param elements
+     */
+    async sendAll(elements: KeyValue[]): Promise<void> {
+        try {
+            await this.inner.sendAll(elements)
+            return
+        } catch (error) {
+            throw new Error(
```

Comment:
why catching instead of throw directly?

Reply:
Great question. I'm not exactly sure, I just followed the pattern of what `sendRecord` did. Maybe it's so that the exception that gets thrown has a more specific error message to the situation?

Reply:
@simlay what do you think?

Reply:
Huh. Yeah, I looked at those `try-catch` blocks and they predate me working on this. I think we don't need them and can just let the errors from the rust propagate up.